source/common/api/BUILD

```diff
@@ -25,11 +25,17 @@ envoy_cc_library(
         "//bazel:linux_x86_64": ["os_sys_calls_impl_linux.cc"],
         "//bazel:linux_aarch64": ["os_sys_calls_impl_linux.cc"],
         "//conditions:default": [],
+    }) + select({
```

Comment:
I used a simpler select here than the one defined in https://github.com/envoyproxy/envoy/blob/619dcbe469d17114ab5c9b079f46866f0f5ce296/bazel/envoy_build_system.bzl#L659 because I didn't see the need to select for apple here if that is taken care of in the hot_restart targets. However, I can rework to use the same build rule here.

Reply:
update: now that I broke up the mocks I can just use the predefined rule. yay for consistency.